### PR TITLE
Add Xero OAuth2 Provider

### PR DESCRIPTION
**Pull request by schmoove**
As per

https://developer.xero.com/documentation/oauth2/overview

### CODE REVIEW

File: src/Xero/Provider.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        $tks = explode('.', $idToken);
+        list($headb64, $bodyb64, $cryptob64) = $tks;
+        $jwtDecoded = JWT::jsonDecode(JWT::urlsafeB64Decode($bodyb64),true);
+
+        return (new User())->map([
+            'id'       => $jwtDecoded->xero_userid,
+            'nickname' => $jwtDecoded->given_name,
+            'name'     => $jwtDecoded->given_name . ' ' . $jwtDecoded->family_name,
+            'email'    => $jwtDecoded->email,
+            'tenant_id' => $connections[0]['tenantId']
```
Comment (atymic) [inline src/Xero/Provider.php:73]: should we not include all of these?

File: src/Xero/Provider.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $connections)
+    {
+        $idToken = $this->credentialsResponseBody['id_token'];
+        $tks = explode('.', $idToken);
```
Comment (atymic) [inline src/Xero/Provider.php:64]: Can you point to the docs for this procress please?

**Comment by atymic** [comment]
Also, please PR to the docs repo.

**Comment by atymic** [comment]
Closing due to inactivity. Ping me if you'd like to continue the PR :)

**Comment by schmoove** [comment]
Sorry have been held up with other work this week. Keen to finish it up and get it merged.

**Comment by atymic** [comment]
No worries, re-opened :)

File: src/Xero/Provider.php (hunk 1 above)
Comment (lukecurtis93) [reply src/Xero/Provider.php:73]: agree with the above - either we get them all in the scope of this process or allow developers to maintain this logic/flow inside their controllers/service classes once they're bounced back to their application (more often than not once tenantIds are retrieved additional logic will be called)

File: src/Xero/Provider.php (hunk 2 above)
Comment (lukecurtis93) [reply src/Xero/Provider.php:64]: https://developer.xero.com/documentation/oauth2/sign-in #4 here :)

File: src/Xero/Provider.php (hunk 1 above)
Comment (lukecurtis93) [inline src/Xero/Provider.php:73]:
```suggestion
            'tenants' => $connections
```

**Comment by atymic** [comment]
ping @schmoove

**Comment by schmoove** [comment]
Happy to do it. Will make that change now.

**Comment by schmoove** [comment]
Will update these today.

> Please add the provider to the split script
> Also, please PR to the docs repo.

**Comment by atymic** [comment]
I did them after the merge, all good :)

**Comment by atymic** [comment]
Tagged as 3.0
https://github.com/SocialiteProviders/Xero/tree/v3.0.0

**Comment by lukecurtis93** [comment]
thanks @atymic  are you able to let us know when this will appear on packigist for composer? https://packagist.org/packages/socialiteproviders/?query=socialiteproviders%2Fxero cheers dude appreciate your speediness of this one :)

**Comment by atymic** [comment]
Oops forgot that step sorry :)
https://packagist.org/packages/socialiteproviders/xero

**Pull request merged**